designer/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentService.scala

```diff
@@ -136,6 +134,7 @@ class DefaultComponentService(
     componentsDefinition.toList
       .map { case (info, definition) =>
         // TODO: We should add componentId into ComponentDefinitionWithImplementation, thx to that we won't need to use ComponentIdProvider in many places
+        // TODO: alternatively, make sure that SingleComponentConfig.componentId is filled during ComponentDefinitionExtractor.extract and rely on it
```

Comment:
You can already add a method `ComponentStaticDefinition.componentIdUnsafe`, we already have `componentGroupUnsafe` and `iconUnsafe` :) It is a zero-cost job, it will already work.

Reply:
turned out not to be so zero-cost, mostly due to the need to fix tests, still, i think it's turned out not too bad